Assets/Content/Items/Functional/Tools/TableConstructer.cs

```diff
@@ -31,10 +34,26 @@ public void Interact()
 
             var tile = targetTile.Tile;
 
-            if (tile.fixture == tableToConstruct) // Deconstruct
-                tile.fixture = null;
-            else // Construct
-                tile.fixture = tableToConstruct;
+            // The player using this item.
+            var player = transform.root;
+
+            // Cancel interaction if the target tile is outside the build range.
+            if (Vector3.Distance(player.transform.position, targetTile.transform.position) > buildDistance)
+            {
+                return;
+            }
```

Comment:
This case should be in `CanInteract` instead. Note: `CanInteract` is ALWAYS called before `Interact`, and `Interact` will only be called if `CanInteract` returns true.